### PR TITLE
v2

### PR DESCRIPTION
**Pull request by revmischa**
* Add auth API for login and token refresh (with tests)
* Nicer init_views function
* `flask config` command to check config
* X-Ray support
* VPC and RDS DB ready to go
* Nice DB base model class, soft deletable, upsertable, UUID, default query filters
* DB user seed
* black

### CODE REVIEW

File: TEMPLATE/db/soft_deletable.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+        return self.filter(self.entity.deleted_on.is_(None))
+
+    def get_filter(self, obj: "SoftDeletable") -> bool:
+        return not obj.deleted_on
+
+
+class SoftDeletable(DefaultQueryFilter):
+    default_filters = [SoftDeletableQueryFilter]
+
+    deleted_on = Column(DateTime(timezone=True))
```
Comment (itiievskyi) [inline TEMPLATE/db/soft_deletable.py]: why not using 
```python
deleted = Column(TSTZ, nullable=True)
```
?

File: TEMPLATE/test/test_auth.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from TEMPLATE.db.fixtures import DEFAULT_PASSWORD
+
+
+def test_login(client_unauthenticated, user, db_session):
+    response = client_unauthenticated.post(
+        "/api/auth/login", json=dict(email=user.email, password=DEFAULT_PASSWORD)
+    )
+    assert response.status_code == 200
+    assert response.json.get("access_token") is not None
```
Comment (itiievskyi) [inline TEMPLATE/test/test_auth.py]: maybe just use
```python
assert response.json.get("access_token")
```
so you can check for both None and empty string?

File: TEMPLATE/db/soft_deletable.py (hunk 1 above)
Comment (revmischa) [reply TEMPLATE/db/soft_deletable.py]: was lazy and didn't want to deal with circular imports. db/__init__ imports this file and it defines TSTZ.

File: TEMPLATE/test/test_auth.py (hunk 2 above)
Comment (revmischa) [reply TEMPLATE/test/test_auth.py]: yeah that would be better

**Pull request merged**